modules/3d/src/rgbd/pose_graph.cpp

```diff
@@ -472,6 +479,112 @@ bool PoseGraphImpl::isValid() const
     return isGraphConnected && !invalidEdgeNode;
 }
 
+double PoseGraphImpl::calculateWeight(const PoseGraphImpl::Edge& e) const
+{
+    double translationNorm = cv::norm(e.pose.t);
+
+    cv::Matx33d R = e.pose.q.toRotMat3x3(cv::QUAT_ASSUME_UNIT);
+    cv::Vec3d rvec;
+    cv::Rodrigues(R, rvec);
+    double rotationAngle = cv::norm(rvec);
+
+    // empirically determined
+    double lambda = 0.485;
```

Comment:
Looks like some data specific parameter. I propose to add lambda parameter with default value 0.485 to `initializePosesWithMST` to have opportunity to change it without OpenCV modification and rebuild. The value may be promoted to the weight calculator as parameter too.